crates/goose/src/agents/agent.rs

```diff
@@ -830,6 +831,34 @@ impl Agent {
         session: Session,
         cancel_token: Option<CancellationToken>,
     ) -> Result<BoxStream<'_, Result<AgentEvent>>> {
+        // Load hint files and add to system prompt
+        let config = Config::global();
```

Comment:
I don't think we should do this in reply_internal - that one is big enough as is. instead add this functionality to the prompt-manager itself and then add it to the call to the prompt manager:

        let prompt_manager = self.prompt_manager.lock().await;
        let system_prompt = prompt_manager
            .builder(model_name)
            .with_extensions(extensions_info.into_iter())
            .with_frontend_instructions(self.frontend_instructions.lock().await.clone())
            .with_extension_and_tool_counts(extension_count, tool_count)
            .build();